### DISCUSSION TITLE
How do you set the color of a solid in iModeljs?

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#4553] [Q&A] [answered]
Hello - How do you set the color of a solid geometry object (sphere, cone etc) in an iModel?
I have tried the following, but it doesn't work. The solids are always white in the iModel?
```
    switch (shape.shapeType) {
      case 'Sphere': {
        const radius = shape.sphereRadius * scaleFactor;
        iModelShape = Sphere.createCenterRadius(Point3d.create(0, 0, radius), radius);
        break;
      }
    }

    // Build the iModel geometry object
    const builder = new GeometryStreamBuilder();
    const params = new GeometryParams("shape");
    params.fillColor = ColorDef.blue; 
    params.lineColor = ColorDef.blue; 
    builder.appendGeometryParamsChange(params);
    if (iModelShape) builder.appendGeometry(iModelShape);
    return builder.geometryStream;
```

**Comment by bbastings** [comment]
> const params = new GeometryParams("shape");

"shape" is not a valid categoryId so I would suspect that as being the issue.

**Reply by PeterJBassett** [reply]
Hello @bbastings - unfortunately the iModel documentation is not very easy to use, so I don't know what a categoryid is or how to define a valid categoryid? 

So please could you explain how to set a valid categoryid?

**Comment by bbastings** [comment, marked as answer]
You are creating a geometric element right? You want to supply the category from your GeometricElementProps.

See [Category](https://www.itwinjs.org/bis/guide/fundamentals/categories/) for more information.

**Reply by PeterJBassett** [reply]
Hello @bbastings - thanks for the link, I wasn't able to find it by searching.

**Comment by pmconne** [comment]
`categoryId` is an `Id64String` which is a 64-bit integer element Id written as a string in hexadecimal like "0x123456".
You will need to query the iModel to find the Id of a category named "shape".
[queryCategoryIdByName](https://www.itwinjs.org/reference/core-backend/categories/spatialcategory/#querycategoryidbyname) can help with that.

**Reply by PeterJBassett** [reply]
@pmconne thank you for your help also.
